crates/trigger-http/src/lib.rs

```diff
@@ -391,6 +386,40 @@ impl HttpTrigger {
         });
     }
 
+    async fn instrumented_service_fn(
+        self: Arc<Self>,
+        addr: SocketAddr,
+        request: Request<Incoming>,
+    ) -> Result<Response<HyperOutgoingBody>> {
+        let span = tracing::info_span!(
+            "spin_trigger_http.handle_http_request",
+            "otel.kind" = "server",
+            "http.request.method" = %request.method(),
+            "network.peer.address" = %addr.ip(),
+            "network.peer.port" = %addr.port(),
+            "network.protocol.name" = "http",
+            "url.path" = request.uri().path(),
+            "url.query" = request.uri().query().unwrap_or(""),
+            "url.scheme" = request.uri().scheme_str().unwrap_or(""),
+            "client.address" = request.headers().get("x-forwarded-for").and_then(|val| val.to_str().ok()),
+            // Recorded later
+            "error.type" = Empty,
+            "http.response.status_code" = Empty,
+            "http.route" = Empty,
+            "otel.name" = Empty,
+        );
+        self.handle(
+            request.map(|body: Incoming| {
+                body.map_err(wasmtime_wasi_http::hyper_response_error)
+                    .boxed()
+            }),
+            Scheme::HTTP,
+            addr,
+        )
+        .instrument(span)
+        .await
```

Comment:
This used to be wrapped in an `async move {}` which I removed. Not sure why it was there.